frontend/ui/test/widgets/trip_view_widget_test.dart

```diff
@@ -21,11 +35,18 @@ void main() {
 
   testWidgets('Showing the expanded Place correctly shows up',
       (WidgetTester tester) async {
-    var tripViewWidget = TripViewWidget();
+    var tripService = MockTripService();
+    var tripCompleter = Completer<Trip>();
+    when(tripService.getTrip(any)).thenAnswer((_) => tripCompleter.future);
+
+    var tripViewWidget = TripViewWidget(tripService, 'id1');
+
     await tester.pumpWidget(wrapForDirectionality(tripViewWidget));
     await tester.pumpAndSettle();
 
-    expect(find.text("Name of the Trip"), findsOneWidget);
+    tripCompleter.complete(Trip(id: 'id1', name: 'name1'));
```

Comment:
I'm guessing you'll want to call `pumpAndSettle()` some time _after_ this line to ensure that the UI picks up its state after loading. @KoopaKing can probably confirm.

Reply:
Actually, given that you aren't testing the pre-load behavior, you can get rid of the `Completer` altogether and just create a future in-line as `Future.value(Trip(id: 'id1', name: 'name1'))`.